Marlin/src/HAL/STM32/u8g_com_stm32duino_ssd_i2c.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2023 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+
+/**
+ * 2 wire I2C COM driver
+ *
+ * Handles both hardware I2C and software I2C.
+ *
+ * The WIRE library is used for the hardware I2C.  That means that
+ * ANY hardware SDA and SLC pairs can be selected and the WIRE library
+ * will automatically enable the correct hardware.
+ *
+ * The SoftWire library is used for the software I2C. That means that
+ * ANY pin can be selected for SDA and SLC.
+ *
+ * This driver requires the SDA and SLC pins be named:
+ *   DOGLCD_SDA
+ *   DOGLCD_SCL
+ *
+ * This allows independence from other I2C devices (mostly EEPROMs) that
+ * usually use I2C_SDA_PIN and I2C_SLC_PIN.
+ *
+ * The following MUST be present in the platformio environment used to
+ * compile Marlin. This required because
+ *
+ * The flag LCD_I2C_SOFT is used to select which I2C library is used.
+ * It must be enabled in the platformio environment used to compile
+ * Marlin.  Enabling it anywhere else results in LCD_I2C_SOFT not being
+ * enabled during crucial portions of the compile process which results
+ * in defaulting to hardwired operation.  The net result is the following
+ * MUST be in the platformio environment in order to use the software I2C
+ * system:
+ *    build_flags = -DLCD_I2C_SOFT
+ *    lib_deps    = stevemarple/SoftWire@^2.0.9
+ *                  stevemarple/AsyncDelay@^1.1.2
+ *
+ */
+
+#include "../../inc/MarlinConfigPre.h"
+
+#if (defined(ARDUINO_ARCH_STM32) && (defined(U8GLIB_SH1106) || defined(IS_U8GLIB_SSD1306) || defined(U8GLIB_SSD1309)))
+
+#include <U8glib-HAL.h>
+
+#include "../../MarlinCore.h"  // so can get SDA & SCL pins
+
+/*
+  BUFFER_LENGTH is defined in libraries\Wire\utility\WireBase.h
+  Default value is 32
+  Increate this value to 144 to send U8G_COM_MSG_WRITE_SEQ in single block
+*/
+
+#if !defined(BUFFER_LENGTH) || BUFFER_LENGTH >= 144
```

Comment:
If increased to 144 as suggested in the comment , the buffer length will be reset to 32